### PR TITLE
feat:adds basic navigation component

### PR DESCRIPTION
**Pull request by FarhanMobashir**
nav
-adds a basic navigation component which is fully responsive
-adds navigation component to the doc site
-made some minor tweak in the badge component

### CODE REVIEW

File: src/components/navigation.css [hunk 1]
```diff
@@ -0,0 +1,117 @@
+.navigation-container {
+  display: flex;
+  justify-content: space-between;
+  /* align-items: center; */
```
Comment (shuklahimanshu81) [inline src/components/navigation.css]: remove comment

Comment (stutisk) [inline src/components/navigation.css]: can remove this comment

File: src/components/navigation.css [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  font-size: var(--tx-18);
+  text-decoration: none;
+  color: var(--black-400);
+}
+
+.nav-icon-container {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+}
```
Comment (shuklahimanshu81) [inline src/components/navigation.css:43]: repeating style. Try using a common class.

**Review by stutisk — APPROVED** [review]
everything else looks good to me !,

File: src/components/navigation.css [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  }
+  .search-input-small {
+    width: 100%;
+    border: none;
+    outline: none;
+    background-color: var(--black-000);
+  }
+  .navigation-link-container {
+    display: none;
+  }
```
Comment (shuklahimanshu81) [inline src/components/navigation.css:115]: Same suggestion here too, repeating style so try using common class.

File: src/components/navigation.css (hunk 1 above)
Comment (FarhanMobashir) [reply src/components/navigation.css]: sure

**Review by shuklahimanshu81 — COMMENTED** [review]
Changes suggested. Good work.

File: src/components/navigation.css (hunk 3 above)
Comment (FarhanMobashir) [reply src/components/navigation.css:115]: here the use case is not generic hence i am not using a common class.

**Pull request merged**